src/io/flutter/run/daemon/FlutterAppService.java

```diff
@@ -45,17 +44,19 @@ private FlutterAppService(Project project) {
   /**
    * Start a Flutter app using the current Flutter SDK.
    * @param workDir The directory where the command should be run.
+   * @param programArgs Additional program args.
    * @param device The device to use.
    * @param mode The RunMode to use (release, debug, profile).
    * @param path Path to the Dart file containing the main method.
    */
   public FlutterApp startFlutterSdkApp(@NotNull String workDir,
+                                       @Nullable String programArgs,
                                        @Nullable FlutterDevice device,
                                        @NotNull RunMode mode,
                                        @Nullable String path)
     throws ExecutionException {
 
-    final GeneralCommandLine command = createFlutterSdkRunCommand(workDir, device, mode, path);
+    final GeneralCommandLine command = createFlutterSdkRunCommand(workDir, programArgs, device, mode, path);
```

Comment:
I won't have the time to do this just now; maybe you or I could follow up with another quick change?  (Need to context switch shortly 😞 )